In [60]:
%matplotlib inline
import pandas as pd

In [3]:
# from IPython.core.display import HTML
# css = open('style-table.css').read() + open('style-notebook.css').read()
# HTML('<style>{}</style>'.format(css))

In [61]:
from google.colab import files
uploaded = files.upload()

Saving titles.csv to titles.csv


In [64]:
import io
titles = pd.read_csv(io.BytesIO(uploaded['titles.csv']))

titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

### How many movies are listed in the titles dataframe?

In [65]:
len(titles)

244914

244914 movies are listed in the titles dataframe

### What are the earliest two films listed in the titles dataframe?

In [66]:
titles.sort_values(by=['year']).head(2)


title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

Miss Jerry and The Startled Lover are the two earliest films. 

### How many movies have the title "Hamlet"?

In [67]:
Hamlet = titles[titles['title']=='Hamlet']
len(Hamlet)


20

20 movies with the title 'Hamlet'

### How many movies are titled "North by Northwest"?

In [68]:
NbN = titles[titles['title']=='North by Northwest']
len(NbN)


1

1 movie with the title 'North by Northwest'

### When was the first movie titled "Hamlet" made?

In [69]:
Hamlet.sort_values(by=['year']).head(1)

title  year
52521  Hamlet  1910

First movie titled "Hamlet" was made in 1910

### List all of the "Treasure Island" movies from earliest to most recent.

In [72]:
TI = titles[titles['title']=='Treasure Island'].sort_values(by=['year'])
TI

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [73]:
movies_1950 = titles[titles['year']==1950]
len(movies_1950)

1113

1113 movies were made in 1950

### How many movies were made in the year 1960?

In [74]:
movies_1960 = titles[titles['year']==1960]
len(movies_1960)

1549

1549 movies were made in 1960

### How many movies were made from 1950 through 1959?

In [75]:
movies_1950_59 = titles[(titles['year']>=1950) & (titles['year']<1960) ] 
len(movies_1950_59)

12934

12934 movies were made from 1950 to 1959

### In what years has a movie titled "Batman" been released?

In [76]:
Batman = titles[titles['title']=='Batman']
Batman.sort_values(by=['year'])

title  year
39349   Batman  1943
240896  Batman  1989

Movie titled batman was released in 1943 and 1989

In [77]:
from google.colab import files
uploaded = files.upload()

In [5]:
import io

casts = pd.read_csv(io.BytesIO(uploaded['cast.csv']))

casts.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

### How many roles were there in the movie "Inception"?

In [7]:
Inception = casts[casts['title']=='Inception']
len(Inception)

78

78 roles in Inception

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [12]:
Inception['n'].isna().sum()

27

27 roles in the movie "Inception" are NOT ranked by an "n" value

### But how many roles in the movie "Inception" did receive an "n" value?

In [13]:
len(Inception) - Inception['n'].isna().sum()

51

51 roles in the movie "Inception" did receive an "n" value

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [16]:
NbN_cast = casts[casts['title']=='North by Northwest']
NbN_cast.sort_values(by=['n']).dropna()

title  year  ...                    character     n
871426   North by Northwest  1959  ...           Roger O. Thornhill   1.0
3518333  North by Northwest  1959  ...                  Eve Kendall   2.0
1457165  North by Northwest  1959  ...              Phillip Vandamm   3.0
3157232  North by Northwest  1959  ...              Clara Thornhill   4.0
356707   North by Northwest  1959  ...                The Professor   5.0
3048272  North by Northwest  1959  ...                Mrs. Townsend   6.0
1695173  North by Northwest  1959  ...              Lester Townsend   7.0
1275337  North by Northwest  1959  ...                      Leonard   8.0
2447081  North by Northwest  1959  ...                     Valerian   9.0
1811387  North by Northwest  1959  ...              Victor Larrabee  10.0
665768   North by Northwest  1959  ...                        Licht  11.0
2295714  North by Northwest  1959  ...                   Auctioneer  12.0
464736   North by Northwest  1959  ...                    Dr. Cross  13.0
1508344  North by Northwest  1959  ...               Sergeant Flamm  14.0
206471   North by Northwest  1959  ...               Captain Junket  15.0
1384780  North by Northwest  1959  ...  Charley - Chicago Policeman  16.0

[16 rows x 6 columns]

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [20]:
Sleuth = casts[(casts['title']=='Sleuth') & (casts['year']==1972)]
Sleuth.sort_values(by=['n'])

title  year  ...                   character    n
1705394  Sleuth  1972  ...                 Andrew Wyke  1.0
326773   Sleuth  1972  ...                 Milo Tindle  2.0
373876   Sleuth  1972  ...           Inspector Doppler  3.0
1465483  Sleuth  1972  ...  Detective Sergeant Tarrant  4.0
2724553  Sleuth  1972  ...             Marguerite Wyke  5.0
1448690  Sleuth  1972  ...      Police Constable Higgs  6.0

[6 rows x 6 columns]

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [21]:
Sleuth = casts[(casts['title']=='Sleuth') & (casts['year']==2007)]
Sleuth.sort_values(by=['n'])

title  year                 name     type          character    n
326774   Sleuth  2007        Michael Caine    actor             Andrew  1.0
1293634  Sleuth  2007             Jude Law    actor               Milo  2.0
1805907  Sleuth  2007        Harold Pinter    actor        Man on T.V.  3.0
260242   Sleuth  2007      Kenneth Branagh    actor  Other Man on T.V.  NaN
373877   Sleuth  2007  Alec (II) Cawthorne    actor  Inspector Doppler  NaN
2724552  Sleuth  2007    Eve (II) Channing  actress    Marguerite Wyke  NaN
3370424  Sleuth  2007    Carmel O'Sullivan  actress             Maggie  NaN

### How many roles were credited in the silent 1921 version of Hamlet?

In [24]:
len(casts[(casts['title']=='Hamlet') & (casts['year']==1921)])

9

9 roles were credited in the silent 1921 version of Hamlet

### How many roles were credited in Branagh’s 1996 Hamlet?

In [25]:
len(casts[(casts['title']=='Hamlet') & (casts['year']==1996)])

54

54 roles were credited in Branagh’s 1996 Hamlet

### How many "Hamlet" roles have been listed in all film credits through history?

In [26]:
len(casts[(casts['title']=='Hamlet')])

342

342 "Hamlet" roles have been listed in all film credits through history

### How many people have played an "Ophelia"?

In [27]:
len(casts[(casts['character']=='Ophelia')])

117

117 people have played an "Ophelia"

### How many people have played a role called "The Dude"?

In [28]:
len(casts[(casts['character']=='The Dude')])

19

19 people have played a role called "The Dude"

### How many people have played a role called "The Stranger"?

In [29]:
len(casts[(casts['character']=='The Stranger')])

212

212 people have played a role called "The Stranger"

### How many roles has Sidney Poitier played throughout his career?

In [30]:
len(casts[(casts['name']=='Sidney Poitier')])

43

Sidney Poitier has played 43 roles throughout his career

### How many roles has Judi Dench played?

In [31]:
len(casts[(casts['name']=='Judi Dench')])

55

Judi Dench has played 55 roles throughout his career

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [33]:
casts[(casts['name']=='Cary Grant') & (casts['year']>=1940) & (casts['year']<1950) & (casts['n']==2)].sort_values(by='year')

title  year        name   type    character    n
871423  My Favorite Wife  1940  Cary Grant  actor   Nick Arden  2.0
871433    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [34]:
casts[(casts['name']=='Cary Grant') & (casts['year']>=1940) & (casts['year']<1950) & (casts['n']==1)].sort_values(by='year')

title  ...    n
871448               The Howards of Virginia  ...  1.0
871407                       His Girl Friday  ...  1.0
871450                The Philadelphia Story  ...  1.0
871438                             Suspicion  ...  1.0
871452                  The Talk of the Town  ...  1.0
871429                 Once Upon a Honeymoon  ...  1.0
871398                     Destination Tokyo  ...  1.0
871422                             Mr. Lucky  ...  1.0
871430                      Once Upon a Time  ...  1.0
871390                  Arsenic and Old Lace  ...  1.0
871425             None But the Lonely Heart  ...  1.0
871424                         Night and Day  ...  1.0
871427                             Notorious  ...  1.0
871444      The Bachelor and the Bobby-Soxer  ...  1.0
871445                     The Bishop's Wife  ...  1.0
871421  Mr. Blandings Builds His Dream House  ...  1.0
871402          Every Girl Should Be Married  ...  1.0
871411                I Was a Male War Bride  ...  1.0

[18 rows x 6 columns]

### How many roles were available for actors in the 1950s?

In [51]:
actor1950s = casts[(casts['year']>=1940) & (casts['year']<1950) & (casts['type']=='actor')]

In [52]:
len(actor1950s.n.dropna().unique())

82

82 roles were available for actors in the 1950s

### How many roles were available for actresses in the 1950s?

In [53]:
actress1950s = casts[(casts['year']>=1940) & (casts['year']<1950) & (casts['type']=='actress')]

In [54]:
len(actress1950s.n.dropna().unique())

76

76 roles were available for actresses in the 1950s

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [56]:
len(casts[(casts['n']==1) & (casts['year']<1980)])

63957

63957 leading roles (n=1) were available from the beginning of film history through 1980

### How many non-leading roles were available through from the beginning of film history through 1980?

In [57]:
len(casts[(casts['n']!=1) & (casts['year']<1980)])

1095378

1095378 non-leading roles were available through from the beginning of film history through 1980?

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [58]:
df1 = casts[(casts['year']<1980)]

In [59]:
df1['n'].isna().sum()

443881

443881 roles through 1980 were minor enough that they did not warrant a numeric "n" rank